### ISSUE TITLE
Components: Render 40px size variants in editor sidebar

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#41973]
Our current priority on the component size consistency front is to get everything in the editor sidebar to be rendered in the 40px height variant. Specifically, this applies to:

- Standard hook-based tools (Typography, Dimensions, Border, etc.) in the edit-post and edit-site editors
- Global Styles sidebar
- `InspectorControlsFill` content in edit-post and edit-site editors

@mtias indicated that it would be best to first validate the new 40px designs in context, as soon as possible. Sizing consistency work on the component library as a whole (#39397) will be put on hold while we work on this first.

## Plan

The previous plan was to leverage the Context System, but after working on this I realized that this is not very feasible, at least in the initial stages. The main problem is that there are just too many components that haven't been hooked up to the context system. There are also a lot of `block-editor` package components being used in the sidebar, and it feels like overkill to connect them all to the context system.

<details>
<summary>Previous plan based on Context System</summary>
<p>

We will try and leverage the [Context System](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/ui/context) so all components of a given type under certain context trees can be rendered in the 40px size by default. Configuration will be added on a [per-component basis](https://github.com/WordPress/gutenberg/pull/41913/files?w=1#diff-bd0fad45336a6b5ca615e2e57337b483ff42ab4d0e50dc9d7248fc050dcecd2bR12).

- [ ] #41913

We need to be careful about how this config cascades. For example, if we suddenly start to render all `InputControl`s at 40px, many higher-level components that depend on `InputControl` will look broken. Therefore, we'll want to add config for higher-level components first.

</p>
</details>

Instead, I'm going to start by flipping the switch section by section in the standard sections (Typography, Dimensions, etc.), to minimize jarring inconsistencies. We can start with the Global Styles section before moving on to the block editor Inspector, since the sections in Global Styles are more isolated visually. Once everything is ready, we can reconsider using the Context System to upsize custom third-party inspector content.

#### If the component does NOT have a default size of 40px:

Create a 40px size variant. (`size = "__unstable-large"`)

#### If the component already has a default size of 40px:

No need to do anything at this time — we'll just leave it be for now. (At some point in the future we will need to decide on a consistent sizing scale across the component library though.)

**Comment by mirka** [comment]
Update: The previous version of this plan used to mention the possibility of a `__unstableSize: 'large'` prop, but I discovered that would create a lot of unnecessary complication when passing through props to underlying components. We don't actually have a `__unstableSize` prop in our codebase yet, so let's just unify with `size: '__unstable-large'`.

**Comment by ciampo** [comment]
Just flagging for completeness that there may be other components that already use the `size` prop to tweak slightly different aspects ([`Card`](https://wordpress.github.io/gutenberg/?path=/story/components-card--default) comes to mind). We may want to review the usage of the `size` prop across the codebase to uniform it as we work on introducing more size variants?

**Comment by mirka** [comment]
`size` prop auditing will be tracked here #42792

**Comment by jasmussen** [comment]
Hey @mirka and @ciampo, I'm coming back to this after a while. Thank you for the patience on this one. 

I've caught up a bit on the conversations around 36px and 40px, and it definitely is difficult to thread the needle with all the moving parts. But at the core, it would be nice to just go straight to 40px for the inspector, as that is what the majority of the designs are for.

Just to visualize it a bit, I put together some quick mockups showing how that would look, with the main gist hopefully being: it looks good! Here's Paragraph, Image, Group:
<img width="364" alt="Paragraph, 40px" src="https://user-images.githubusercontent.com/1204802/182346305-fe5f422e-6391-47a6-bb6d-8bfd678766cc.png">

<img width="364" alt="Image, 40px" src="https://user-images.githubusercontent.com/1204802/182346315-18801f34-ed1b-4bd5-ab97-0028efd2ac15.png">

<img width="364" alt="Group, 40px" src="https://user-images.githubusercontent.com/1204802/182551654-bcff5ddd-9014-42f5-9fe8-6483091b5de8.png">

Please note that the mockups above are meant mainly to illustrate the 40px height across inputs, controls, and buttons, but they are likely **slightly out of date in other aspects**, notably around the Group layout panel, which I know is being explored separately. 

Let me know if this is useful, and how I can best help move things forward! Nice work.

**Comment by mirka** [comment]
Update: I changed my plan — the Context System plan turned out to be kind of impractical at this stage. See updated OP for specifics.

**Comment by mirka** [comment]
Closing in favor of #46741
